### PR TITLE
Linux imx 4.14.98

### PR DESCRIPTION
**Pull request by zeeblex**
Update to 4.14.98 kernel, kernel headers and related modules.

### CODE REVIEW

File: recipes-kernel/linux/linux-imx_4.14.98.bb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        mkdir -p ${B}
+        cp ${S}/arch/arm/configs/imx_v7_defconfig ${B}/.config
+        cp ${S}/arch/arm/configs/imx_v7_defconfig ${B}/../defconfig
+    else
+        # copy latest defconfig to use for mx8
+        mkdir -p ${B}
+        cp ${S}/arch/arm64/configs/defconfig ${B}/.config
+        cp ${S}/arch/arm64/configs/defconfig ${B}/../defconfig
+    fi
+}
```
Comment (otavio) [inline recipes-kernel/linux/linux-imx_4.14.98.bb]: Please drop this custom code.

File: recipes-kernel/linux/linux-imx_4.14.98.bb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+DESCRIPTION = "Linux Kernel provided and supported by NXP with focus on \
+i.MX Family Reference Boards. It includes support for many IPs such as GPU, VPU and IPU."
+
+require recipes-kernel/linux/linux-imx.inc
+require linux-imx-src-${PV}.inc
+
+DEPENDS += "lzop-native bc-native"
+
+DEFAULT_PREFERENCE = "1"
+
```
Comment (otavio) [inline recipes-kernel/linux/linux-imx_4.14.98.bb]: This is the default if we use linux-imx, drop it.

File: recipes-kernel/linux/linux-imx-headers_4.14.98.bb [hunk 3]
```diff
@@ -7,11 +7,11 @@ New headers are installed in ${includedir}/imx."
...
 LIC_FILES_CHKSUM = "file://COPYING;md5=d7810fab7487fb0aad327b76f1be7cd7"
 
-SRCBRANCH = "imx_4.9.123_imx8mm_ga"
-LOCALVERSION = "-imx"
-SRC_URI = "git://source.codeaurora.org/external/imx/linux-imx.git;protocol=https;branch=${SRCBRANCH} \
-    file://0001-uapi-Install-custom-headers.patch"
-SRCREV = "6a71cbc089755afd6a86c005c22a1af6eab24a70"
+SRCBRANCH = "imx_4.14.98_2.0.0_ga"
+LOCALVERSION = "-${SRCBRANCH}"
+KERNEL_SRC ?= "git://source.codeaurora.org/external/imx/linux-imx.git;protocol=https"
```
Comment (otavio) [inline recipes-kernel/linux/linux-imx-headers_4.14.98.bb]: Please avoid new variables.

File: recipes-kernel/linux/linux-imx-headers_4.14.98.bb [hunk 4]
```diff
@@ -55,5 +54,3 @@ do_install() {
 ALLOW_EMPTY_${PN} = "1"
 
 PACKAGE_ARCH = "${MACHINE_SOCARCH}"
-COMPATIBLE_HOST_imx = ".*"
-COMPATIBLE_HOST ?= "(none)"
```
Comment (otavio) [inline recipes-kernel/linux/linux-imx-headers_4.14.98.bb]: This is required.

File: recipes-kernel/linux/linux-imx-headers_4.14.98.bb (hunk 3 above)
Comment (zeeblex) [reply recipes-kernel/linux/linux-imx-headers_4.14.98.bb]: Done!

**Comment by angolini** [comment]
@otavio frindly reminder

**Pull request merged**